stdlib/src/builtin/string.mojo

```diff
@@ -930,11 +989,26 @@ struct String(
     fn _as_ptr(self) -> DTypePointer[DType.int8]:
         """Retrieves a pointer to the underlying memory.
 
+        Note that you should use `_as_uint8_ptr()` if you need to access the
+        pointer as we are now storing the bytes as UInt8.
+
+        See https://github.com/modularml/mojo/issues/2317 for more information.
+
         Returns:
             The pointer to the underlying memory.
         """
         return rebind[DTypePointer[DType.int8]](self._buffer.data)
 
+    fn _as_uint8_ptr(self) -> DTypePointer[DType.uint8]:
```

Comment:
**Suggestion** Include a `TODO(mojo#2317): Remove this method` comment here so we don't forget to remove this method down the line.

Reply:
Done